[17.0][OU-ADD] product: migrate to 17.0

hbrunn opened the pull request:
https://github.com/OCA/OpenUpgrade/pull/4468 with review followup

haumenphai committed:
[OU-ADD] product: migrate to 17.0

hbrunn committed:
[OU-IMP] product: populate product_document table from attachments

hbrunn commented:
/ocabot migration product

<dependency></dependency>

hbrunn commented:
/ocabot migration product

remi-filament commented on openupgrade_scripts/scripts/product/17.0.1.2/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,91 @@
+
+product      / product.product          / activity_user_id (many2one)   : not related anymore
+product      / product.product          / activity_user_id (many2one)   : now a function
+product      / product.product          / message_main_attachment_id (many2one): DEL relation: ir.attachment
+product      / product.product          / product_document_ids (one2many): NEW relation: product.document
```
Thank you @hbrunn for taking over this work !
You should probably mark these 2 lines as DONE in post-migration instead of NOTHING TO DO

hbrunn replied:
for the first one see my [question](https://github.com/OCA/OpenUpgrade/pull/4600#issuecomment-2427155081) in the mail PR, agreed with the second, done in 074f09d36537 and will squash when you agree

pedrobaeza committed:
[OU-FIX] product: Finish migration scripts

- Preserve old pricelist order.
- Remove the base pricelist when applicable.
- Clean incorrect packaging records.
- Improve the analysis work file explanations.

TT49320

pedrobaeza commented:
I have pushed an extra commit with the following things:

- Preserve old pricelist order.
- Remove the base pricelist when applicable.
- Clean incorrect packaging records.
- Improve the analysis work file explanations.
